### ISSUE TITLE
Misleading error message: Job for httpd.service failed because a configured resource limit was exceeded

### ISSUE THREAD

**Issue by minfrin** [systemd/systemd#2716]
### Submission type

[X] Bug report

[ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> systemd-219-19.el7_2.4.x86_64

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> CentOS7
### In case of bug report: Expected behaviour you didn't see

> When the EnvironmentFile directive is used, and this file is missing, the following error is written to the internal log:

Feb 23 12:29:46 localhost systemd[1]: Failed to load environment files: No such file or directory

What I expected to see was the same / similar error reported by the systemctl command on the console.
### In case of bug report: Unexpected behaviour you saw

> The error reported to the systemctl command however is a completely different error:

[root@localhost ~]# systemctl restart httpd
Job for httpd.service failed because a configured resource limit was exceeded. See "systemctl status httpd.service" and "journalctl -xe" for details.
### In case of bug report: Steps to reproduce the problem

> Take a vanilla CentOS7 machine, and move the /etc/sysconfig/httpd file out of the way, then attempt to restart httpd. You'll get the misleading error message.

**Comment by poettering** [comment]
Upstream is currently at 229, please use this form only to submit bugs in the two mostly released systemd versions, which is 229 and 228 hence.

Also, sounds like a packaging bug in httpd anyway.

**Comment by poettering** [comment]
Please report this to your distro's bug tracker, thanks.

**Comment by mbiebl** [comment]
@poettering the bug reporter mentioned a misleading error message when an environment file is missing. This is easily reproducible with v229 as well. Adding a non-existing EnvironmentFile= to a service file leads to a rather cryptic error message like this:

```
 systemctl restart NetworkManager
Job for NetworkManager.service failed because a configured resource limit was exceeded. See "systemctl status NetworkManager.service" and "journalctl -xe" for details.
```

I added a bogus EnvironmentFile=/does/not/exist to NetworkManager.service

**Comment by minfrin** [comment]
Just to confirm, this bug is related to the misleading error message in systemd, it bears no relationship to httpd whatsoever.

Bug reported upstream to CentOS:

https://bugs.centos.org/view.php?id=10432

**Comment by johannbg** [comment]
This bug is valid and confirmed on systemd-229-2.fc24.x86_64 

Adding EnvironmentFile=/path/bogus fails
Job for dmscad.service failed because a configured resource limit was exceeded. See "systemctl status dmscad.service" and "journalctl -xe" for details.
systemd[1]: dmscad.service: Failed to load environment files: No such file or directory
systemd[1]: dmscad.service: Failed to run 'start' task: No such file or or directory
systemd[1]: Failed to start Tivoly Storage Manager Client.

On top of that there seem to be more inconsistencies in failure behaviour. 

For example just adding bogus entry to the [Service] section like so
[Service]
bogus

results in 
Failed to start dmscad.service: Unit dmscad.service is not loaded properly: Invalid argument.
systemd[1]: [/etc/systemd/system/dmscad.service:6] Missing '='.

Adding just that missing '=' as in 
[Service]
bogus=
Ignores the line 
"systemd[1]: [/etc/systemd/system/dmscad.service:6] Unknown lvalue 'bogus' in section 'Service' 
and continue starting the service 

Arguably in all these three cases they all should have failed ( with a non cryptic error message ) or all lines should have been ignored and the service continued to be started.

**Comment by poettering** [comment]
Fix is waiting in #3119.
